### PR TITLE
Refactor out old API interactions for new Reliably Entity Server

### PR DESCRIPTION
**Pull request by ciaransweet**
# What I am changing

TL;DR: this PR removes the old Reliably API probes and tolerances in favour of using the new Reliably Entity Server. It adds `get_objective_results_by_labels` as a probe which takes a `limit` and a dict of `labels` which match the Objectives labels that we're using for our SSH. In combination with the tolerance `all_objective_results_ok`, the user can determine if their SLO was impacted from running an experiment.

**Sorry it's a bit big** 😇

# How I did it

* Replaced `requests` with `httpx`
* `chaosreliably/__init__.py`:
  * Removed `get_user` and `get_default_org` as they are now redundant
  * Refactored `get_auth_info` to accommodate the users' organisation as well
* `chaosreliably/slo/probes.py`:
  * Removed `get_last_N_slos`, `get_slo_history`, and `fetch_history`
  * Added `get_objective_results_by_labels` which queries for the objective results of a given objective
* `chaosreliably/slo/tolerances.py`:
  * Remove `last_N_slo_were_met_for_all_services`
  * Added `all_objective_results_ok` which takes a list of Objective Results and determines if they were OK or not
* Removed `chaosreliably/slo/types.py` as this wasn't being used
* Add dependencies on:
  * `black`, `isort`, `flake8`, and `pytest-httpx` for development (formatting & mocking httpx)
  * `httpx` for core package usage to replace `requests`
* Removed dependencies:
  * `pylama`, `requests_mock`, and `pycodestyle` as they were no longer needed
* Tests:
  * Added `conftest.py` to provide some fixtures for the tests
  * Added tests for the new probe and tolerance
  * Refactored `test_auth.py` to include `org`
* Updated the README
* Added `Makefile` to abstract away linting & formatting
* Switched `.github/workflows/build.yaml` to use `make lint` for linting checks

# How you can test it

To check that the changes have tests and work _within_ the project, first setup for development as per the Contributing section of the README, then run:

```bash
$ pytest
```

Or you can get setup with Reliably and install `chaostoolkit-reliably` and use it in your experiments:

image

### CODE REVIEW

File: chaosreliably/__init__.py [hunk 1]
```diff
@@ -1,52 +1,36 @@
...
-                secrets: Secrets = None) -> requests.Session:
-    host, token = get_auth_info(configuration, secrets)
+def get_session(
+    configuration: Configuration = None, secrets: Secrets = None
+) -> httpx.Client:
+    auth_info = get_auth_info(configuration, secrets)
     headers = {
         "Content-Type": "application/json",
-        "Authorization": "Bearer {}".format(token)
+        "Authorization": "Bearer {}".format(auth_info["token"]),
```
Comment (Lawouach) [inline chaosreliably/__init__.py:26]: should we use f-strings everywhere? is it supported from Python 3.6? if not, let's fall back to "".format(...)" indeed.

File: chaosreliably/__init__.py [hunk 2]
```diff
@@ -65,56 +50,77 @@ def discover(discover_system: bool = True) -> Discovery:
...
-            reliably_config_path))
+        logger.debug("Loading Reliably config from: {}".format(reliably_config_path))
         with open(reliably_config_path) as f:
             try:
                 config = yaml.safe_load(f)
             except yaml.YAMLError as ye:
                 raise ActivityFailed(
                     "Failed parsing Reliably configuration at "
-                    "'{}': {}".format(reliably_config_path, str(ye)))
+                    "'{}': {}".format(reliably_config_path, str(ye))
```
Comment (Lawouach) [inline chaosreliably/__init__.py:81]: likewise on the f-string

File: chaosreliably/__init__.py [hunk 3]
```diff
@@ -65,56 +50,77 @@ def discover(discover_system: bool = True) -> Discovery:
...
+        current_org = config.get("currentOrg")
+        if current_org:
+            reliably_org = current_org.get("name")
+
+    if (
+        not reliably_config_path
+        and not reliably_token
+        and not reliably_host
+        and not reliably_org
+    ):
```
Comment (Lawouach) [inline chaosreliably/__init__.py:99]: see this is wher I dislike black... sigh :D I have to let go

**Review by Lawouach — APPROVED** [review]
LGTM

In the future, may I request more focused PRs. One for the linting, one for the refactoring (removing the types module for instance) and one for the main change. That makes things a bit easire to review and read from the git log in the future. Since you haven't squashed your commits accordingly, the history won't show clearly these tasks.

File: chaosreliably/__init__.py (hunk 1 above)
Comment (ciaransweet) [reply chaosreliably/__init__.py:26]: @Lawouach as of 3.6 f-strings are around 😄

File: chaosreliably/__init__.py (hunk 3 above)
Comment (ciaransweet) [reply chaosreliably/__init__.py:99]: Better one opinion to rule them all, than many opinions across files 🤣
